std/process.d

```diff
@@ -416,15 +418,21 @@ private Pid spawnProcessImpl(in char[][] args,
 
     int[2] forkPipe;
     if (core.sys.posix.unistd.pipe(forkPipe) == 0)
-    {
         setCLOEXEC(forkPipe[1], true);
-    }
     else
-    {
         throw ProcessException.newFromErrno("Could not create pipe to check startup of child");
-    }
     scope(exit) close(forkPipe[0]);
 
+    int[2] pidPipe;
+    if (config & Config.detached)
+    {
+        if (core.sys.posix.unistd.pipe(pidPipe) == 0)
+            setCLOEXEC(pidPipe[1], true);
+        else
+            throw ProcessException.newFromErrno("Could not create pipe to get process pid");
+    }
+    scope(exit) if (config & Config.detached) close(pidPipe[0]);
```

Comment:
I guess calling `getpid` from the second child is also a possibility, but I guess it's not worth the change.